### PR TITLE
ConfigCLIOptions::configFile: --config relative paths are relative to…

### PR DESCRIPTION
**Pull request by fvacek**
… the

location of the program binary #217

### CODE REVIEW

File: libshvcore/src/utils/clioptions.cpp [hunk 1]
```diff
@@ -400,53 +404,62 @@ bool ConfigCLIOptions::loadConfigFile()
...
+		else if(is_absolute_path(config_dir))
+			return make_tuple(config_dir, joinPath(config_dir, default_config_name));
+		else
+			return make_tuple(joinPath(cwd, config_dir), joinPath(cwd, config_dir, default_config_name));
+	}
+	else if(is_absolute_path(config_file)) {
+		if(config_dir.empty()) {
+			auto sep_pos = config_file.find_last_of('/');
+			// absolute config file must contain '/'
+			assert(sep_pos != std::string::npos);
```
Comment (syyyr) [inline libshvcore/src/utils/clioptions.cpp:447]: You already check for `is_absolute_path(config_file)`, doesn't that imply that there's a slash at the beginning? Also what if `config_file == "/"`? That's not a valid file, can you add a test for that?

Comment (fvacek) [reply libshvcore/src/utils/clioptions.cpp:447]: Then config dir will be empty and config file will be `/`, we will probably get exception later on opening config file what is IMO sufficient.

The assert() is superfluous, I'll remove it.

Comment (fvacek) [reply libshvcore/src/utils/clioptions.cpp]: yes

Comment (fvacek) [reply libshvcore/src/utils/clioptions.cpp]: test for missing slash or for config == `/`

Comment (syyyr) [reply libshvcore/src/utils/clioptions.cpp]: For config == /

Comment (fvacek) [reply libshvcore/src/utils/clioptions.cpp]: and what should I return in this case `("", "")`?

Comment (syyyr) [reply libshvcore/src/utils/clioptions.cpp]: I guess if you're fine with "/" returning just "/", then "" should return ""

Comment (fvacek) [reply libshvcore/src/utils/clioptions.cpp]: I'm still in doubts about handling the special case like `/`, what about `//` or `.`. I thing that we should solve all or nothing.

Comment (syyyr) [reply libshvcore/src/utils/clioptions.cpp]: I think the correct solution is to error out early (or at least according to what I've read in multiple places on the internet, e.g. here https://softwareengineering.stackexchange.com/a/419918).

I'm not exactly sure where to do that. Ideally the `CLIOPTION_GETTER_SETTER` should have some sort of an option to supply a validation callback. Then we wouldn't need to think about what `ConfigCLIOptions::absoluteConfigPaths()` should return in invalid cases.

Comment (fvacek) [reply libshvcore/src/utils/clioptions.cpp]: Correct solution is to check that absolute-config-dir and absolute-config-file exist and readable, what is not in scope of function `ConfigCLIOptions::absoluteConfigPaths()`. 

I cannot make some good test in this function and I also do not like to add one particular test for '/', so I suggest catch invalid config file/dir later.

Comment (syyyr) [reply libshvcore/src/utils/clioptions.cpp]: I agree that this is not the scope of this function.

I think that the invalid config file/dir should be checked even before this function is called. But that's another discussion/patch/something. :)

I'm resolving this.

**Pull request merged**